spec/data-types-and-values.html

```diff
@@ -7,12 +7,59 @@ <h1>ECMAScript Language Types</h1>
 
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
-      <p>TBD</p>
+      <p>
+        The Record type is the set of all finite mappings from Strings to ECMAScript primitive values including Record and Tuple. Each record value holds an associated [[Fields]] List value which is a list of pairs of the form { [[Key]], [[Value]] } where the [[Key]] is a String and [[Value]] is any primitive value. Entries of [[Fields]] are sorted by [[Key]] in code unit order. The [[Fields]] List and its entries are never modified.
+     </p>
+
+      <emu-clause id="sec-makerecordvalue" aoid="MakeRecordValue">
+        <h1>MakeRecordValue ( _entries_ )</h1>
+        <p>This abstract operation MakeRecordValue takes a _entries_ argument. It is used to specify the creation of new Record value types. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: _entries_ is a List.
+            1. Let _reversedEntries_ be a new List containing the same values as the list _entries_ where the values are ordered as an Array of the same values has been sorted using `Array.prototype.reverse`.
+            1. Let _uniqueEntries_ be an empty List.
+            1. For each _kv_ in _reversedEntries_:
+              1. Assert: Type(_kv_.[[Key]]) is String.
+              1. Assert: Type(_kv_.[[Value]]) is not Object.
+              1. If there is no entry _existing_ in _uniqueEntries_ such that _existing_.[[Key]] is _kv_.[[Key]], append _kv_ to _uniqueEntries_.
+            1. Let _sortedEntries_ be a new List containing the same values as the list _uniqueEntries_ where the values are ordered as if an Array of the same values had been sorted using `Array.prototype.sort` using <emu-xref href="#sec-recordentrycompare">RecordEntryCompare</emu-xref> as _comparefn_.

```

Comment:
This language still needs to change, as I described in https://github.com/tc39/proposal-record-tuple/pull/132#discussion_r445150000 . I understand that you tried to copy it from that place, but it doesn't make sense to apply that text here, as it is sorting strings (which are JS values) and you're sorting records here (which aren't) and passing in algorithm in place of a function (which is also invalid). I suggest you use the wording I provided to start, and we can iterate on it later.

Reply:
Got it!

Reply:
similarly here; delegating to a public method doesn't seem like an ideal approach to me.